application_sandbox/extended_dynamic_state/main.cpp

```diff
@@ -0,0 +1,537 @@
+// Copyright 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+
+#include "application_sandbox/sample_application_framework/sample_application.h"
+#include "mathfu/matrix.h"
+#include "mathfu/vector.h"
+#include "support/entry/entry.h"
+#include "vulkan_helpers/buffer_frame_data.h"
+#include "vulkan_helpers/helper_functions.h"
+#include "vulkan_helpers/vulkan_application.h"
+#include "vulkan_helpers/vulkan_model.h"
+
+using Mat44 = mathfu::Matrix<float, 4, 4>;
+using Vector4 = mathfu::Vector<float, 4>;
+
+namespace cube_model {
+#include "cube.obj.h"
+}
+const auto& cube_data = cube_model::model;
+
+uint32_t cube_vertex_shader[] =
+#include "cube.vert.spv"
+    ;
+
+uint32_t cube_fragment_shader[] =
+#include "cube.frag.spv"
+    ;
+
+VkPhysicalDeviceExtendedDynamicStateFeaturesEXT kExtendedDynamicStateFeatures =
+    {VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_EXTENDED_DYNAMIC_STATE_FEATURES_EXT,
+     nullptr, VK_TRUE};
+
+const VkFormat kDepthStencilFormat = VK_FORMAT_D32_SFLOAT_S8_UINT;
```

Comment:
I know we use this in a number of places, but I just realized this format is not required to be supported.
Either VK_FORMAT_D24_UNORM_S8_UINT  or VK_FORMAT_D32_SFLOAT_S8_UINT must be supported.
We may want to expose a method on VulkanApplication to say what depth stencil format should be used.

Reply:
Added support and query for supported formats in sample application. Only used in this sample for now.